packages/ui-components/src/components/CheckboxList.tsx

```diff
@@ -89,47 +89,45 @@ export interface ListItemProps extends Record<string, unknown> {
   tooltip?: string;
 }
 
-function not(a: ListItemProps[], b: ListItemProps[]): ListItemProps[] {
-  return a.filter(
-    item => b.findIndex(i => i[i.valueKey || 'code'] === item[item.valueKey || 'code']) === -1,
-  );
-}
-
-function intersection(a: ListItemProps[], b: ListItemProps[]): ListItemProps[] {
-  return a.filter(
-    item => b.findIndex(i => i[i.valueKey || 'code'] === item[item.valueKey || 'code']) !== -1,
-  );
-}
-
-function union(a: ListItemProps[], b: ListItemProps[]): ListItemProps[] {
-  return [...a, ...not(b, a)];
-}
-
-interface CheckboxListProps {
-  list: ListItemProps[];
-  title?: string;
-  selectedItems: ListItemProps[];
-  setSelectedItems: (items: ListItemProps[]) => void;
-}
-
-export const CheckboxList = ({
+export const CheckboxList = <ListItem extends ListItemProps>({
   list,
   title = 'Choices',
   selectedItems,
   setSelectedItems,
-}: CheckboxListProps) => {
-  const numberOfChecked = (items: ListItemProps[]) => intersection(selectedItems, items).length;
+}: {
+  list: ListItem[];
+  title?: string;
+  selectedItems: ListItem[];
+  setSelectedItems: (items: ListItem[]) => void;
+}) => {
+  const not = (a: ListItem[], b: ListItem[]) => {
+    return a.filter(
+      item => b.findIndex(i => i[i.valueKey || 'code'] === item[item.valueKey || 'code']) === -1,
+    );
+  };
+
+  const intersection = (a: ListItem[], b: ListItem[]) => {
+    return a.filter(
+      item => b.findIndex(i => i[i.valueKey || 'code'] === item[item.valueKey || 'code']) !== -1,
+    );
+  };
+
+  const union = (a: ListItem[], b: ListItem[]) => {
+    return [...a, ...not(b, a)];
+  };
```

Comment:
Kind of a bummer that we need to move these function in to the component. We could make them generic themselves, but I didn't quite feel like writing so much boilerplate 😅